commands/container-registry/token/tokens.go

```diff
@@ -0,0 +1,184 @@
+	tokenCmd.AddCommand(TokenReplaceCmd())
+	tokenCmd.AddCommand(scope.TokenScopesCmd())
+
+	return tokenCmd
+}
+
+func getTokenPrint(
+	resp *ionoscloud.APIResponse, c *core.CommandConfig, response *[]ionoscloud.TokenResponse,
+	post bool,
+) printer.Result {
+	r := printer.Result{}
+	cols, _ := c.Command.Command.Flags().GetStringSlice(constants.ArgCols)
+
+	if c != nil {
+		if resp != nil {
+			r.Resource = c.Resource
+			r.Verb = c.Verb
+			r.WaitForState = viper.GetBool(
+				core.GetFlagName(
+					c.NS, constants.ArgWaitForRequest,
+				),
+			) // this boolean is duplicated everywhere just to do an append of `& wait` to a verbose message
+		}
+		if response != nil {
+			if !post {
+				defaultHeaders := []string{"TokenId", "DisplayName", "ExpiryDate", "Status"}
+				r.OutputJSON = response
+				r.KeyValue = getTokensRows(response) // map header -> rows
+				r.Columns = printer.GetHeaders(
+					allCols, defaultHeaders, cols,
+				) // headers
+			} else {
+				r.OutputJSON = response
+				r.KeyValue = getTokensRows(response)
+				postHeaders := []string{"DisplayName", "ExpiryDate", "Status"} // map header -> rows
+				r.Columns = printer.GetHeaders(allCols, postHeaders, cols)     // headers
+			}
+		}
+	}
+	return r
+}
+
+type TokenPrint struct {
+	TokenId             string `json:"TokenId,omitempty"`
+	DisplayName         string `json:"DisplayName,omitempty"`
+	ExpiryDate          string `json:"ExpiryDate,omitempty"`
+	CredentialsUsername string `json:"CredentialsUsername,omitempty"`
+	CredentialsPassword string `json:"CredentialsPassword,omitempty"`
+	Status              string `json:"Status,omitempty"`
+}
+
+func getTokensRows(tokens *[]ionoscloud.TokenResponse) []map[string]interface{} {
+	out := make([]map[string]interface{}, 0, len(*tokens))
+	for _, token := range *tokens {
+		var tokenPrint TokenPrint
+		if idOk, ok := token.GetIdOk(); ok && idOk != nil {
+			tokenPrint.TokenId = *idOk
+		}
+		if propertiesOk, ok := token.GetPropertiesOk(); ok && propertiesOk != nil {
+			if displayNameOk, ok := propertiesOk.GetNameOk(); ok && displayNameOk != nil {
+				tokenPrint.DisplayName = *displayNameOk
+			}
+			if expiryDateOk, ok := propertiesOk.GetExpiryDateOk(); ok && expiryDateOk != nil {
+				tokenPrint.ExpiryDate = expiryDateOk.String()
+			}
+			if credentialsOk, ok := propertiesOk.GetCredentialsOk(); ok && credentialsOk != nil {
+				if usernameOk, ok := credentialsOk.GetUsernameOk(); ok && usernameOk != nil {
+					tokenPrint.CredentialsUsername = *usernameOk
+				}
+				if passwordOk, ok := credentialsOk.GetPasswordOk(); ok && passwordOk != nil {
+					tokenPrint.CredentialsPassword = *passwordOk
+				}
+			}
+			if statusOk, ok := propertiesOk.GetStatusOk(); ok && statusOk != nil {
+				tokenPrint.Status = *statusOk
+			}
+		}
+		o := structs.Map(tokenPrint)
+		out = append(out, o)
+	}
+	return out
+}
+
+var allCols = structs.Names(TokenPrint{})
+
+// TODO: check how other resources are doing this
```

Comment:
Other resources take the given parent resource as a param for the completer, and use that to make a GET for the child object. In this case, you'd take the registry ID as a param.

However, this is also fine